syntax = "proto3";
import "google/protobuf/empty.proto";

service SecretStore {

  //ref store calls
  rpc SetSecret(SetSecretRequest) returns (google.protobuf.Empty) {}
  rpc GetSecret(GetSecretRequest) returns (GetSecretResponse) {}
  rpc GetSecrets(GetSecretsRequest) returns (GetSecretsResponse) {}
  rpc DeleteSecret(DeleteSecretRequest) returns (google.protobuf.Empty) {}
}

message SetSecretRequest {
  bytes agent_id = 1;
  string key = 2;
  string value = 3;
}

message GetSecretRequest {
  bytes agent_id = 1;
  string key = 2;
}

message GetSecretResponse{
  bytes agent_id = 1;
  string key = 2;
  optional string value = 3;
}

message GetSecretsRequest {
  bytes agent_id = 1;
  optional string key_prefix = 2;
}

message GetSecretsResponse{
  bytes agent_id = 1;
  map<string, bytes> values = 2;
}

message DeleteSecretRequest {
  bytes agent_id = 1;
  string key = 2;
}

